lib/core/macros.nim

```diff
@@ -798,10 +798,17 @@ proc newLit*[T](s: set[T]): NimNode {.compileTime.} =
     var typ = getTypeInst(typeof(s))[1]
     result = newCall(typ,result)
 
-proc newLit*(arg: tuple): NimNode {.compileTime.} =
-  result = nnkPar.newTree
-  for a,b in arg.fieldPairs:
-    result.add nnkExprColonExpr.newTree(newIdentNode(a), newLit(b))
+proc isNamedTuple(T: typedesc): bool {.magic: "TypeTrait".}
+  ## Return true for named tuples, false for any other type.
```

Comment:
nit: i'd rather have a comment like: `see typetraits.isNamedTuple` to keep things DRY (in case doc comment changes in typetraits.isNamedTuple)